mne/report.py

```diff
@@ -316,6 +316,35 @@ def _update_html(html, report_fname, report_sectionlabel):
     return htmls, report_fnames, report_sectionlabels
 
 
+def open_report(fname, **params):
+    """Read a saved report or, if it doesn't exist yet, create a new one.
+
+    The returned report is used as a context manager and any changes to the
+    report are saved when exiting the context block.
+
+    Parameters
+    ----------
+    fname : str
+        The file containing the report, stored in the HDF5 format. If the file
+        does not exist yet, a new report is created that will be saved to the
+        specified file.
+    **params : list of parameters
+        Parameters to use when creating a new Report object.
+
+    Returns
+    -------
+    report : instance of Report
+        The report.
+    """
+    if op.exists(fname):
+        # Don't pass **params to the Report
+        report = Report(fname=fname)
+        report.__setstate__(read_hdf5(fname, title='mnepython'))
+    else:
+        report = Report(fname=fname, **params)
```

Comment:
why not throw an error if the file does not exist? It seems like this is another way to instantiate a report object ...

Reply:
It is indeed another way to instantiate a report object. 

Often, an analysis pipeline is spread out across multiple scripts (for example, our submissions to the Frontiers Research Topic). Instead of having one script at the very end that generates all the figures and writes them to a report, I'd like an API that makes it easy for each script to append some figures to the report as it runs. When we're constructing and debugging the pipeline, we often run single scripts in isolation.

Hence, it is convenient to have a quick way to open a report or create a new one if it doesn't exist yet.